src/ui/public/chrome/directives/global_nav/app_switcher/app_switcher.js

```diff
@@ -72,6 +72,15 @@ uiModules
       // links don't cause full-navigation events in certain scenarios
       // so we force them when needed
       this.ensureNavigation = appSwitcherEnsureNavigation;
+
+      $scope.getTooltip = link => {
```

Comment:
`$scope` is a perfectly fine way of exposing values to the view, but it has drawbacks when you start involving child scopes (shadowing, unreflected writes, etc.)

Since around angular 1.5 and the new component type, controllers have just been promoted to be the preferred way to expose values to the view as they provide a namespace preventing the inherited values problem, they can be required by child directives/components, and I think they generally describe their purpose better (the controller controls the view, not the `$scope`).